protocol-units/bridge/chains/ethereum/src/lib.rs

```diff
@@ -0,0 +1,294 @@
+		hash_lock: HashLock<Self::Hash>,
+		time_lock: TimeLock,
+		amount: Amount,
+	) -> BridgeContractInitiatorResult<()> {
+		let contract = AtomicBridgeInitiator::new(initiator_address.0 .0, &self.rpc_provider);
+		let recipient_bytes: [u8; 32] = recipient_address.0.try_into().unwrap();
+		let call = contract.initiateBridgeTransfer(
+			U256::from(amount.0),
+			FixedBytes(recipient_bytes),
+			FixedBytes(hash_lock.0),
+			U256::from(time_lock.0),
+		);
+		let _ = send_transaction(
+			call,
+			&self.send_transaction_error_rules,
+			self.num_tx_send_retries,
+			self.gas_limit as u128,
+		)
+		.await;
+		Ok(())
+	}
+
+	async fn complete_bridge_transfer(
+		&mut self,
+		bridge_transfer_id: BridgeTransferId<Self::Hash>,
+		pre_image: HashLockPreImage,
+	) -> BridgeContractInitiatorResult<()> {
+		let pre_image: [u8; 32] = utils::vec_to_array(pre_image.0)?;
+		let contract = AtomicBridgeInitiator::new(self.signing_address.0, &self.rpc_provider);
+		let call = contract
+			.completeBridgeTransfer(FixedBytes(bridge_transfer_id.0), FixedBytes(pre_image));
+		let _ = send_transaction(
+			call,
+			&self.send_transaction_error_rules,
+			self.num_tx_send_retries,
+			self.gas_limit as u128,
+		)
+		.await;
+		Ok(())
+	}
+
+	async fn refund_bridge_transfer(
+		&mut self,
+		bridge_transfer_id: BridgeTransferId<Self::Hash>,
+	) -> BridgeContractInitiatorResult<()> {
+		let contract = AtomicBridgeInitiator::new(self.signing_address.0, &self.rpc_provider);
+		let call = contract.refundBridgeTransfer(FixedBytes(bridge_transfer_id.0));
+		let _ = send_transaction(
+			call,
+			&self.send_transaction_error_rules,
+			self.num_tx_send_retries,
+			self.gas_limit as u128,
+		)
+		.await;
+		Ok(())
+	}
+
+	async fn get_bridge_transfer_details(
+		&mut self,
+		bridge_transfer_id: BridgeTransferId<Self::Hash>,
+	) -> BridgeContractInitiatorResult<Option<BridgeTransferDetails<Self::Address, Self::Hash>>> {
+		let mapping_slot = U256::from(0); // the mapping is the zeroth slot in the contract
+		let key = bridge_transfer_id.0;
+		let storage_slot = self.calculate_storage_slot(key, mapping_slot);
+		let storage: U256 = self
+			.rpc_provider
+			.get_storage_at(self.initiator_address.0, storage_slot)
+			.await
+			.map_err(|_| BridgeContractInitiatorError::GetMappingStorageError)?;
+		let storage_bytes = storage.to_be_bytes::<32>();
+		let mut storage_slice = &storage_bytes[..];
+		let eth_details = EthBridgeTransferDetails::decode(&mut storage_slice)
+			.map_err(|_| BridgeContractInitiatorError::DecodeStorageError)?;
+		let details = BridgeTransferDetails {
+			bridge_transfer_id,
+			initiator_address: InitiatorAddress(eth_details.originator),
+			recipient_address: RecipientAddress(eth_details.recipient.to_vec()),
+			hash_lock: HashLock(eth_details.hash_lock),
+			time_lock: TimeLock(eth_details.time_lock.wrapping_to::<u64>()),
+			amount: Amount(eth_details.amount.wrapping_to::<u64>()),
+		};
+
+		Ok(Some(details))
+	}
+}
+
```

Comment:
You mean on eth side? On movement side it is of course. And all the aptos counterparty code is here in this PR 👍

Reply:
Yes, the implementation of the abstract bridge necessitates completion on both sides concurrently. If only one side is implemented, it will be necessary to separate it.

Reply:
Hmm I thought the whole plan was to stand up the abstract bridge and E2E test it in one direction first. If this is the case, we will need to prioritise the contracts for the other direct. What's your preference?